client/src/lib/query/useTaskMutation.ts

```diff
@@ -24,16 +25,18 @@ function updateActivitiesCache(updatedActivity: ActivityWithIds) {
 				byId: {
 					[updatedActivity.activity_id]: updatedActivity
 				}
-			};
+			} as { byId: ById<ActivityWithIds> };
 
 		if (!updatedActivity) return old;
 
-		return {
+		const newActivitiesData = {
```

Comment:
Don't need to put this into a variable. Think I did it to check the typing in dev.